src/main/java/net/pms/network/HTTPServer.java

```diff
@@ -195,14 +194,19 @@ public void stop() {
 			}
 		}
 
-		if (channel != null) { // HTTP Engine V2
-			if (group != null) {
-				group.close().awaitUninterruptibly();
+		/**
+		 * Netty v3 (HTTP Engine V2) shutdown approach from
+		 * @see https://netty.io/3.8/guide/#start.12
+		 */
+		if (channel != null) {
+			if (allChannels != null) {
+				allChannels.close().awaitUninterruptibly();
 			}
+			LOGGER.info("Confirm allChannels is empty: " + allChannels.toString());
 
-			if (factory != null) {
-				factory.releaseExternalResources();
-			}
+			LOGGER.info("This log happens");
+			bootstrap.releaseExternalResources();
```

Comment:
@SubJunk what about to add some delay before executing this to let the UMS to stop properly all classes?

Reply:
@valib that could work. I would also like to understand why it hangs though since their manual says it should be safe to do in this way. I don't think we are creating new channels in the meantime but maybe I'm wrong.